src/matlab2tikz.m

```diff
@@ -1473,14 +1474,21 @@ case guitypes()
             % axis styles (e.g., colors).
             options = opts_add(options, 'separate axis lines');
         end
```

Comment:
In #880 there are some figures to illustrate the behavior (which is where most of this code stems from).
I'm also not 100% convinced about this, but it matches MATLAB behavior better, which is what is expected from `strict` mode.

However, now I think about it: perhaps it is better to only add this fragment when the strict parameter is passed.